packages/manager/apps/pci-load-balancer/src/pages/create/steps/region/RegionStep.tsx

```diff
@@ -1,145 +1,185 @@
   OsdsSpinner,
   OsdsText,
 } from '@ovhcloud/ods-components/react';
-import { ODS_TEXT_LEVEL, ODS_TEXT_SIZE } from '@ovhcloud/ods-components';
+import {
+  ODS_ICON_NAME,
+  ODS_TEXT_LEVEL,
+  ODS_TEXT_SIZE,
+} from '@ovhcloud/ods-components';
 import { ODS_THEME_COLOR_INTENT } from '@ovhcloud/ods-common-theming';
-import { StepComponent } from '@ovh-ux/manager-react-components';
-import { useTranslation } from 'react-i18next';
-import { ShapesInputComponent } from '@ovh-ux/manager-pci-common';
-import { useEffect, useRef } from 'react';
+import {
+  StepComponent,
+  Links,
+  useProjectUrl,
+} from '@ovh-ux/manager-react-components';
+import { Trans, useTranslation } from 'react-i18next';
+import {
+  RegionSelector,
+  useProject,
+  usePCICommonContextFactory,
+  PCICommonContext,
+} from '@ovh-ux/manager-pci-common';
 import { TRegion } from '@/api/hook/useRegions';
 import { REGION_AVAILABILITY_LINK } from '@/constants';
 import { StepsEnum, useCreateStore } from '@/pages/create/store';
 import { useTracking } from '@/pages/create/hooks/useTracking';
-import { LabelComponent } from './components/Label.component';
-import { StackLabelComponent } from './components/StackLabel.component';
-import { GroupLabelComponent } from './components/GroupLabel.component';
-import { StackTitleComponent } from '@/pages/create/steps/region/components/StackTitle.component';
-import { useColumnsCount } from '@/pages/create/hooks/useColumnsCount';
 
 export type TRegionStepProps = {
   isLoading: boolean;
-  isMobile: boolean;
   regions: Map<string, TRegion[]>;
   ovhSubsidiary: string;
 };
+
+const isRegionWith3AZ = (regions: TRegion[]) =>
+  regions.some((region) => region.type === 'region-3-az');
+
 export const RegionStep = ({
   isLoading,
-  isMobile,
   regions,
   ovhSubsidiary,
 }: Readonly<TRegionStepProps>): JSX.Element => {
-  const { t: tCommon } = useTranslation('pci-common');
-  const { t: tCreate } = useTranslation('load-balancer/create');
-
-  const columnsCount = useColumnsCount();
+  const { t } = useTranslation(['load-balancer/create', 'pci-common']);
+  const { data: project } = useProject();
+  const projectUrl = useProjectUrl('public-cloud');
 
   const { trackStep } = useTracking();
 
   const store = useCreateStore();
 
-  // TODO: remove this once the class is added to step component directly in manager-react-components
-  useEffect(() => {
-    const refEl = document.getElementById('ref');
-    if (refEl) {
-      const stepEl = refEl.nextSibling;
-      if (stepEl) {
-        const targetEl = stepEl.childNodes.item(1);
-
-        if (targetEl.nodeName === 'DIV') {
-          const target = targetEl as HTMLDivElement;
-          if (!target.classList.contains('max-w-[calc(100%-76px)]')) {
-            target.classList.add('max-w-[calc(100%-76px)]');
-          }
-        }
-      }
-    }
-  }, []);
+  const has3AZ = useMemo(() => {
+    const allRegions = regions
+      ? Array.from(regions, ([, values]) => values)
+      : [];
```

Comment:
I am not sure to understand this, could this be a .values() instead ?

Also this could be simplified I think

Reply:
Here I want to convert a `Map`, to an array of regions. IMO it's the fast way and with less line of code. But I'm interested if you have an example using `.values()`

Reply:
@Tsiorifamonjena 
I guess:
```suggestion
  const allRegions = regions ? [...regions.values()] : [];
```

Reply:
Yeah thanks 👌🏽